src/overlays/actors/ovl_Fishing/z_fishing.c

```diff
@@ -216,97 +325,5459 @@ static ColliderJntSphInit D_80B7A888 = {
+    Matrix_MultVec3f(&posSrc, &posStep);
+
+    refPos->x = pos->x + posStep.x;
+    refPos->y = pos->y + posStep.y;
+    refPos->z = pos->z + posStep.z;
+
+    Matrix_Translate(pos->x, pos->y, pos->z, MTXMODE_NEW);
+
+    if ((player->actor.speedXZ == 0.0f) && (D_80B7E138 == 0.0f)) {
+        Math_ApproachF(&sLureHookRotY[hookIndex], ry, 0.1f, 0.3f);
+    } else {
+        sLureHookRotY[hookIndex] = ry;
+    }
+
+    Matrix_RotateY(sLureHookRotY[hookIndex], MTXMODE_APPLY);
+    Matrix_RotateX(rx, MTXMODE_APPLY);
+    Matrix_Scale(0.0039999997f, 0.0039999997f, 0.005f, MTXMODE_APPLY);
+    Matrix_RotateY(M_PI, MTXMODE_APPLY);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_fishing.c", 3029),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, gFishingLureHookDL);
+
+    Matrix_RotateZ(M_PI / 2, MTXMODE_APPLY);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_fishing.c", 3034),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, gFishingLureHookDL);
+
+    if ((hookIndex == 1) && (D_80B7A68C != 0)) {
+        Matrix_Scale(2.0f, 2.0f, 2.0f, MTXMODE_APPLY);
+        Matrix_Translate(250.0f, 0.0f, -1400.0f, MTXMODE_APPLY);
+        Matrix_Push();
+
+        if (D_80B7A690 != 0) {
+            FishingEffect* effect = globalCtx->specialEffects;
+            MtxF mf;
+
+            Matrix_MultVec3f(&sZeroVec, &effect->pos);
+            Matrix_Get(&mf);
+            func_800D20CC(&mf, &sEffOwnerHatRot, 0);
+
+            D_80B7A690 = 0;
+            D_80B7A68C = 0;
+
+            effect->type = FS_EFF_OWNER_HAT;
+            effect->unk_2C = 0;
+            effect->vel = sZeroVec;
+            effect->accel = sZeroVec;
+        }
+
+        Matrix_Pop();
+        Matrix_Translate(-1250.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_RotateX(M_PI / 2, MTXMODE_APPLY);
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_fishing.c", 3085),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, gFishingOwnerHatDL);
+    }
+
+    Matrix_Pop();
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_fishing.c", 3098);
+}
+
+void Fishing_UpdateSinkingLure(GlobalContext* globalCtx) {
+    s16 i;
+    f32 dx;
+    f32 dy;
+    f32 dz;
+    f32 rx;
+    f32 ry;
+    f32 dist;
+    f32 offsetY;
+    Vec3f posSrc = { 0.0f, 0.0f, 0.0f };
+    Vec3f posStep;
+    Vec3f sp94;
+    Vec3f sp88;
+    f32 offsetX;
+    f32 offsetZ;
+    Player* player = PLAYER;
+
+    posSrc.z = 0.85f;
+
+    sSinkingLurePos[0] = sLurePos;
+
+    if (D_80B7A6D4 != 0) {
+        offsetY = -1.0f;
+    } else if (sLurePos.y < WATER_SURFACE_Y(globalCtx)) {
+        offsetY = 0.5f;
+    } else {
+        offsetY = -5.0f;
+    }
+
+    if (D_80B7A694 == 5) {
+        Matrix_RotateY(player->actor.shape.rot.y * (M_PI / 32768), MTXMODE_NEW);
```

Comment:
```suggestion
        Matrix_RotateY(player->actor.shape.rot.y * (M_PI / 0x8000), MTXMODE_NEW);
```
same in other places

Reply:
I don't know how I feel about using 0x8000 for floats, especially when we have to cast them.
Ideally these conversions should be done through macros, but they clearly weren't consistent about it so we can't use the same macro everywhere. For instance `(foo / 32768.0f) * M_PI` and `foo * (M_PI / 32768.0f)` are both used in the file and equivalent, but they don't compile the same.